deployments/salt/splunk-otel-collector/config.sls

```diff
@@ -0,0 +1,24 @@
+{% set splunk_otel_collector_config = salt['pillar.get']('splunk-otel-collector:splunk_otel_collector_config', '/etc/otel/collector/agent_config.yaml') %}
+
+{% set splunk_otel_collector_config_source = salt['pillar.get']('splunk-otel-collector:splunk_otel_collector_config_source') %}
+
+{% set splunk_service_user = salt['pillar.get']('splunk-otel-collector:service_user', 'splunk-otel-collector') %}
+
+{% set splunk_service_group = salt['pillar.get']('splunk-otel-collector:service_group', 'splunk-otel-collector') %}
+
+{% if splunk_otel_collector_config_source  != '' %}
+
+Push custom config for Splunk Otel Collector, if provided:
+  file.managed:
```

Comment:
Please make sure the collector service is restarted if the config file has changed.

Reply:
@jcheng-splunk 
We tried using the `Watch` module for restarting the collector service however it's only working with files changed using Salt, So added the `file.copy` module to copy the default config. This ensures that the service will restart if the config file and environment file get changed.